openapi3filter/issue625_test.go

```diff
@@ -0,0 +1,242 @@
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/getkin/kin-openapi/openapi3filter"
+	"github.com/getkin/kin-openapi/routers/gorillamux"
+)
+
+func TestIssue625(t *testing.T) {
+
+	anyItemSpec := `
+openapi: 3.0.0
+info:
+  version: 1.0.0
+  title: Sample API
+paths:
+ /items:
+  get:
+    description: Returns a list of stuff
+    parameters:
+    - description: test object
+      explode: false
+      in: query
+      name: test
+      required: false
+      schema:
+        type: array
+        items: {}   ###
+    responses:
+      '200':
+        description: Successful response
+`[1:]
+
+	objectArraySpec := `
+openapi: 3.0.0
+info:
+  version: 1.0.0
+  title: Sample API
+paths:
+ /items:
+  get:
+    description: Returns a list of stuff
+    parameters:
+    - description: test object
+      explode: false
+      in: query
+      name: test
+      required: false
+      schema:
+       type: array
+       items:
+         type: object
+         properties:
+           name:
+            type: string
+    responses:
+      '200':
+        description: Successful response
+`[1:]
+
+	anyOfArraySpec := `
+openapi: 3.0.0
+info:
+  version: 1.0.0
+  title: Sample API
+paths:
+ /items:
+  get:
+    description: Returns a list of stuff
+    parameters:
+    - description: test object
+      explode: false
+      in: query
+      name: test
+      required: false
+      schema:
+        type: array
+        items:
+         anyOf:
+          - type: integer
+          - type: boolean
+    responses:
+      '200':
+        description: Successful response
+`[1:]
+
+	allOfArraySpec := `
+openapi: 3.0.0
+info:
+  version: 1.0.0
+  title: Sample API
+paths:
+ /items:
+  get:
+    description: Returns a list of stuff
+    parameters:
+    - description: test object
+      explode: false
+      in: query
+      name: test
+      required: false
+      schema:
+        type: array
+        items:
+         allOf:
+          - type: integer
+          - type: number
+    responses:
+      '200':
+        description: Successful response
+`[1:]
+
+	oneOfArraySpec := `
+openapi: 3.0.0
+info:
+  version: 1.0.0
+  title: Sample API
+paths:
+ /items:
+  get:
+    description: Returns a list of stuff
+    parameters:
+    - description: test object
+      explode: false
+      in: query
+      name: test
+      required: false
+      schema:
+        type: array
+        items:
+         oneOf:
+          - type: integer
+          - type: boolean
+    responses:
+      '200':
+        description: Successful response
+`[1:]
+
+	tests := []struct {
+		name  string
+		spec  string
+		req   string
+		isErr bool
+	}{
+		{
+			name:  "successful any item array",
+			spec:  anyItemSpec,
+			req:   "/items?test=3",
+			isErr: false,
+		},
+		{
+			name:  "successful any item object array",
+			spec:  anyItemSpec,
+			req:   `/items?test={"name": "test1"}`,
```

Comment:
This doesn't seem to be url-encoded. Are you sure this is a valid URL?

Reply:
You are right. This is not a valid URL. 
Lets focus the fix on array with items of polymorphism type  (i.e anyOf, oneOf, allOf ). This is the needed use case.
I removed the test case and relevant code